src/main/java/sysc4806/project24/mini_survey_monkey/interceptors/SurveyStateInterceptor.java

```diff
@@ -0,0 +1,50 @@
+package sysc4806.project24.mini_survey_monkey.interceptors;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import sysc4806.project24.mini_survey_monkey.models.State;
+import sysc4806.project24.mini_survey_monkey.models.Survey;
+import sysc4806.project24.mini_survey_monkey.repositories.SurveyRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class SurveyStateInterceptor implements HandlerInterceptor {
+
+    private final SurveyRepository surveyRepository;
+    private List<State> stateToFilter = new ArrayList<>();
+
+    public SurveyStateInterceptor(SurveyRepository surveyRepository) {
+        this.surveyRepository = surveyRepository;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        Object surveyIDAttribute = request.getAttribute("surveyID");
+        if (surveyIDAttribute == null) {
+            return true;
+        }
```

Comment:
**Why does this return `true` if `surveyID` is null?**

When I read this, my gut says this should return false. I think it returns `true` because intercepting a `null` `surveyID` is outside the responsibility of this class.

**Could you add an inline comment explaining why we return true?**

Reply:
It's actually because the `/create` endpoint to create a new survey doesn't have a surveyID attribute, so visiting this would result in an error.

You could ask the exact same question about the logic below that, because that returns true when no survey is found for a surveyID. I left it like that for the reason you said - I think it should be the responsibility of the class making the POST request to handle the error.

I will add comments explaining both!